### DISCUSSION TITLE
Black Screen on all Clients

### DISCUSSION THREAD

**Discussion by coryknuth** [ClassicOldSong/Apollo#175] [Q&A]
Hello All,

Just switched to Apollo from Sunshine and it's fantastic work, trying to get fully comfortable though. I had it working great, on all clients, and I enabled virtual display, headless mode, and set the display settings to disconnect my two monitors in Windows when the stream starts. Leaving only the virtual display. It was working perfectly and would reactivate the physical monitors on stream end, but then randomly, after many successful stream starts and endings, I now only get a black screen on all clients when I start a stream, sometimes a mouse cursor appears but I cannot move it. And it still disables my main monitors so I can't see the display settings when I start stream. Is there a setting somewhere I am missing? Should I reinstall apollo from scratch?

Any advice appreciated,
Thanks!

**Comment by ClassicOldSong** [comment]
Does rebooting your computer fix the problem somehow?

Have you upgraded your graphics card driver recently?

**Reply by coryknuth** [reply]
Rebooting didn't clear it,  haven't tweaked the driver since switching from Sunshine.  I was able to get to the Apollo config through the app (handy feature) and when I disabled headless mode I was able to get the monitors on and reset the display settings.  Now as long as I choose virtual display for the app,  it works,  as long as headless mode is off. So maybe I just misunderstood what that option does?

**Reply by ClassicOldSong** [reply]
Headless mode is intended for headless systems without any display connected, or hybrid configurations like laptops with a dGPU. Basically it skips initial display detection and only detect after a connection is established and the virtual display is created.

It should be paired with "Adapter Name" settings when your system has two or more graphics cards, which enforce the virtual display to use the specific graphics card you specified.

**Reply by Atsoup** [reply]
I had a similar issue earlier while setting up Apollo. The virtual display would show a black screen with no cursor or anything. I enabled Headless Mode, and it works without any issues whatsoever. I rebooted the computer to see if it would fix the issue, but it persisted. Only after enabling Headless Mode is when it works for me.

**Reply by ClassicOldSong** [reply]
Set Adapter Name and try again.

The problem happens because when not in headless mode, the initial probing already set Apollo to run on iGPU, but the virtual display was created on the dGPU. So image cannot be shared between the two GPUs directly, resulting a black image being sent.
